File: readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int words_calc(string text, int len);
float l_calc(string text, int len, int word);
float s_calc(string text, int len, int word);
void result(float letters, float sentence);

int main(void)
{
    string input = get_string("Text: ");
    int len = strlen(input);

    // calculating words
    int words = words_calc(input, len);

    // calculating l
    float l = l_calc(input, len, words);

    // calculating s = average number of sentences per 100 words
    float s = s_calc(input, len, words);

    // result
    result(l, s);
}

int words_calc(string text, int len)
{
    int n = 0;
    for (int i = 0; i <= len; i++)
    {
        if (text[i] == 32)
        {
            n += 1;
        }
        if (i == len)
        {
            n += 1;
        }
    }
    return n;
}

float l_calc(string text, int len, int word)
{
    int n = 0;
    for (int i = 0; i <= len; i++)
    {
        if ((text[i] <= 90 && text[i] >= 65) || (text[i] <= 122 && text[i] >= 97))
        {
            n += 1;
        }
    }

    // calculaating l = average number of letters in per 100 words
    float l = (n / (float) word) * 100.00;
    return l;
}

float s_calc(string text, int len, int word)
{
    int n = 0;
    for (int i = 0; i <= len; i++)
    {
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            n += 1;
        }
    }
    float s = (n / (float) word) * 100.00;
    return s;
}

void result(float letters, float sentence)
{
    // calculating readebility by Coleman-Liau index
    float index = 0.0588 * letters - 0.296 * sentence - 15.8;
    int grade = round(index);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: sentimental-readability/readability.py
def main():
    text = input("Text: ")
    grade = int(round(calc(text)))
    if grade < 1:
        print("Before Grade 1...")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {int(round(calc(text)))}")


def calc(text):
    letters = 0
    words = 0
    sentences = 0
    index = 0
    for char in text:
        if char.isalpha():
            letters += 1
        if char == ' ':
            words += 1
        if (char == '.' or char == '?' or char == '!') and text[index - 1].isalpha():
            sentences += 1
        index += 1
    if letters > 0:
        words += 1
    grade = grading(letters, words, sentences)
    return grade


def grading(letters, words, sentences):
    l = l_calc(letters, words)
    s = s_calc(sentences, words)
    return (0.0588 * l - 0.296 * s - 15.8)


def l_calc(letters, words):
    return (100 * letters) / words


def s_calc(sentences, words):
    return (100 * sentences) / words

main()
